src/app/search/search.controller.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Created by Ilya on 04/11/15.
+ */
+
+//var _Search = new WeakMap();
+
+class SearchController {
+    constructor(Search) {
+        this.query = '';
+        this.repos = [];
+        this.Search = Search;
+    }
+
+    searchRepos() {
+        if (this.query) {
+            this.Search.getRepositories({q: this.query}, (data) => {
```

Comment:
What if first request will take 10 seconds and second request will take 1 second? What will user see after 10 seconds?